### PR TITLE
Tag lifecycle fix and styling updates

### PR DESCRIPTION
**Pull request by j1mie**
Updates tag component so that icons can be changed via props. 

Also updates to use the actionIcon component.

Moves component changes out of componentDidMount / state and into a function called in render.

#### Changelog

**New**

- TagUpdate example in story
- buildAction fn 
- Lightwrapper component for story
- Rounded modifier for actionIcon

**Changed**

- Replaces icons with actionIcon
- Uses flex to layout inner elements
- Pass in props for action functions
- Pass in props for icon descriptions

**Removed**

- LeftAction prop
- Some extra SASS modifiers

### CODE REVIEW

File: src/components/Tag/Tag-story.js [hunk 1]
```diff
@@ -6,16 +6,53 @@ import { Tag, TagWrapper } from '../../index';
...
+            Offering Managment &middot; {this.state.count}
+          </Tag>
+        </TagWrapper>
+      </LightWrapper>
+    );
+  }
+}
+
 const actionTagProps = {
-  leftAction: true,
```
Comment (j1mie) [inline src/components/Tag/Tag-story.js]: @stpCollabr8nLstn Removed this as I couldn't find its usage - that OK?

File: src/components/Tag/_tag.scss [hunk 2]
```diff
@@ -78,46 +76,33 @@
...
+  }
 }
 
-.#{$prefix}--tag--left__remove,
-.#{$prefix}--tag--left__add {
-  fill: $text-01;
-
-  &:hover {
-    fill: $brand-01;
-    .#{$prefix}--tag--left__st {
```
Comment (j1mie) [inline src/components/Tag/_tag.scss]: @stpCollabr8nLstn Could not find this being used but happy to reinsert if it's inadvertently breaking something!

File: src/components/Tag/_tag.scss [hunk 3]
```diff
@@ -78,46 +76,33 @@
...
 }
 
 .#{$prefix}--tag--left__success {
-  fill: $brand-01;
-}
-
-.#{$prefix}--tag--left__st {
-  stroke: $text-01;
-  stroke-width: 1;
+  .#{$prefix}--action-icon_img {
```
Comment (j1mie) [inline src/components/Tag/_tag.scss]: I know, a bit of a BEM anti-pattern here!

File: src/components/Tag/Tag-story.js [hunk 4]
```diff
@@ -27,13 +64,15 @@ storiesOf('Components|Tag', module)
...
       `,
     )(() => (
-      <TagWrapper aria-label="Tags">
-        <Tag className="some-class">Offering Managment &middot; 72</Tag>
-        <Tag className="some-class">Development &middot; 63</Tag>
-        <Tag className="some-class">RTP &middot; 51</Tag>
-        <Tag className="some-class">Littleton &middot; 33</Tag>
-        <Tag className="some-class">Dublin &middot; 12</Tag>
-      </TagWrapper>
+      <LightWrapper>
```
Comment (j1mie) [inline src/components/Tag/Tag-story.js]: @stpCollabr8nLstn Should I add the background to `TagWrapper` instead?

File: src/components/Tag/Tag-story.js (hunk 1 above)
Comment (stpCollabr8nLstn) [reply src/components/Tag/Tag-story.js]: 👍

File: src/components/Tag/_tag.scss (hunk 2 above)
Comment (stpCollabr8nLstn) [reply src/components/Tag/_tag.scss]: 👍

File: src/components/Tag/_tag.scss (hunk 3 above)
Comment (stpCollabr8nLstn) [reply src/components/Tag/_tag.scss]: I'll allow it

File: src/components/Tag/Tag-story.js (hunk 4 above)
Comment (stpCollabr8nLstn) [reply src/components/Tag/Tag-story.js]: yeah - that'd be cleaner

File: src/components/Tag/_tag.scss (hunk 3 above)
Comment (j1mie) [reply src/components/Tag/_tag.scss]: Thanks A!

File: src/components/Tag/Tag-story.js (hunk 4 above)
Comment (j1mie) [reply src/components/Tag/Tag-story.js]: Will do!

**Pull request merged**